osi3trace/osi_trace.py

```diff
@@ -49,13 +50,16 @@ def from_file(self, path, type_name="SensorView", cache_messages=False):
         self.message_offsets = [0]
         self.message_cache = {} if cache_messages else None
 
+        self.timestep_count = len(self.retrieve_offsets())
```

Comment:
This turns a simple open operation into a very expensive operation, as it must read through the whole file to extract all the offsets. Sometimes this is needed, but in most instances you want to actually read and process a file sequentially, either step-by-step, or you need all messages anyway, in which case you also do not want to retrieve the offsets but rather read the messages at the same time. So that is a big no-no.